services/applicantService.js

```diff
@@ -1,12 +1,130 @@
+      beginning_time: route.beginning_time,
+      ending_date: route.ending_date,
+      ending_time: route.ending_time,
+      plane_needed: route.plane_needed,
+      hotel_needed: route.hotel_needed
+    }))
+  ];
+};
+
+// Function to calculate the total number of days from the routes
+export const getRequestDays = (routes) => {
+  if (!routes || routes.length === 0) return 0;
+
+  // Sort routes by router_index
+  const sortedRoutes = routes.sort((a, b) => a.router_index - b.router_index);
+
+  const firstRoute = sortedRoutes[0];
+  const lastRoute = sortedRoutes[sortedRoutes.length - 1];
+
+  // Combine date and time strings to create full datetime objects
+  const startDate = new Date(`${firstRoute.beginning_date}T${firstRoute.beginning_time}`);
+  const endDate = new Date(`${lastRoute.ending_date}T${lastRoute.ending_time}`);
+
+  // Calculate the difference in milliseconds
+  const diffInMs = endDate - startDate;
+
+  // Convert milliseconds to days (with decimal)
+  const diffInDays = diffInMs / (1000 * 60 * 60 * 24);
+
+  // Use ceil to always round up if there's any partial day
+  const dayDiff = Math.ceil(diffInDays);
+
+  return dayDiff;
+};
+
+export const cancelTravelRequestValidation = async (request_id) => {
+  try {
+    const status_id = await Applicant.getRequestStatus(request_id);
+    if (status_id === null) {
+      throw { status: 404, message: "Travel request not found" };
+    }
+
+    if (![1, 2, 3, 4, 5].includes(status_id)) {
+      throw {
+        status: 400,
+        message: "Request cannot be cancelled after reaching 'Atención Agencia de Viajes'"
+      };
     }
 
-    const insertedCount = await Applicant.createExpenseBatch(receipts);
-    return insertedCount;
-  }
\ No newline at end of file
+    await Applicant.cancelTravelRequest(request_id);
+
+    return {
+      message: "Travel request cancelled successfully",
+      request_id,
+      request_status_id: 9,
+      active: false
+    };
+  } catch (err) {
+    console.error("Error in cancelTravelRequest service:", err);
+    throw err;
+  }
+};
+
+export async function createExpenseValidationBatch(receipts) {
+  if (!Array.isArray(receipts) || receipts.length === 0) {
+    const err = new Error('The "receipts" field must be a non-empty array');
+    err.code = 'BAD_REQUEST';
+    throw err;
+  }
+
+  const insertedCount = await Applicant.createExpenseBatch(receipts);
+  return insertedCount;
+}
+
+// Check if the country exists in the database If not, insert it
+export const getCountryId = async (conn, countryName) => {
+  console.log("Checking country:", countryName);
+  const countryQuery = `SELECT country_id FROM country WHERE country_name = ?`;
+  const [CountryRows] = await conn.query(countryQuery, [countryName]);
+  //If country does not exist, insert it
+  if (CountryRows === undefined) {
+    console.log("Country not found, inserting:", countryName);
+    const insertCountryQuery = `INSERT INTO country (country_name) VALUES (?)`;
```

Comment:
```suggestion
    const insertCountryQuery = `INSERT INTO Country (country_name) VALUES (?)`;
```
Another incorrect name for the Country table.